### PR TITLE
fix extra files not being included in write-manifest

### LINKED ISSUE #416: write_manifest include extra files does not work when executed outside of target directory
Related to https://github.com/rstudio/rsconnect-python/issues/411

write_manifest include extras is bugged at the moment when we try to include the extra file from outside the folder:

```
mkdir app
touch app/.env app/main.py
pip install rsconnect-python==1.16.0
rsconnect write-manifest streamlit app app/.env

# Results:
> Checking arguments...                            [OK]
> Inspecting Python environment...                 [OK]
> Creating manifest.json...                        [ERROR]
Error: ../.env must be under app.
```

The temporary workaround by cd into the app directory before executing the command:

```
cd app
rsconnect write-manifest streamlit ./ ./env
```

### PR DESCRIPTION
**Pull request by bcwu**
## Intent

fixes #416 

## Type of Change

- [x] Bug Fix            
- [ ] New Feature       
- [ ] Breaking Change   

## Checklist

- [x] I have updated [CHANGELOG.md](../CHANGELOG.md) to cover notable changes.
- [x] I have updated all related GitHub issues to reflect their current state.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,19 +5,13 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
...
 ### Fixed
-
 - cacert read error when adding/updating a server.
-
-## Unreleased
-
-### Added
-- Added `system caches list` and `system caches delete` commands which allow administrators to enumerate and delete R and Python runtime caches from Connect servers.
+- getdefaultlocale deprecated
+- extra files not being included in write-manifest
```
Comment (mmarchetti) [inline CHANGELOG.md:14]: It would be helpful to add issue numbers and better descriptions to these entries.

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,19 +5,13 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 ## Unreleased
 
-### Fixed
-- getdefaultlocale deprecated
-
-## Unreleased
+### Added
+- Added `system caches list` and `system caches delete` commands which allow administrators to enumerate and delete R and Python runtime caches from Connect servers.
```
Comment (mmarchetti) [inline CHANGELOG.md:9]: This should probably be added to the PR that introduces those new features (along with documentation for them).

File: rsconnect/main.py [hunk 3]
```diff
@@ -1816,7 +1816,6 @@ def _write_framework_manifest(
 
     with cli_feedback("Checking arguments"):
         entrypoint = validate_entry_point(entrypoint, directory)
-        extra_files = validate_extra_files(directory, extra_files)
```
Comment (mmarchetti) [inline rsconnect/main.py:1819]: Can you explain the approach being taken here? Is this validation being done at a later stage in the bundle creation/manifest writing?

Comment (bcwu) [reply rsconnect/main.py:1819]: This change prevents extra_files from being processed twice. Once here (now deleted) and another time in https://github.com/rstudio/rsconnect-python/blob/2adcec54dc8bbb1bea77f2a1967510ec69044919/rsconnect/bundle.py#L1804

**Comment by bcwu** [comment]
This would be easier to fix with PurePath.is_relative_to, but that is only available in Python 3.9

**Comment by kgartland-rstudio** [comment]
On the new build, I'm still hitting the same error:

```
PS C:\Users\User> rsconnect version
1.16.1.dev119+g69411ba
```
File listing:
```
PS C:\Users\User> ls .\voila\


    Directory: C:\Users\User\voila


Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
d-----         2/16/2023   7:33 AM                empty-folder
d-----         2/16/2023   7:35 AM                extra-folder
d-----         2/16/2023   7:38 AM                packrat
d-----         2/16/2023   7:41 AM                rsconnect-python
-a----         3/23/2023   8:20 AM      104857600 bigtest
-a----         2/16/2023   7:27 AM            160 environment.yml
-a----         2/16/2023   7:27 AM           8260 index.ipynb
-a----         2/16/2023   7:27 AM           1553 LICENSE
-a----         3/23/2023   8:12 AM        1048576 newtest
-a----         3/24/2023   6:44 AM             49 requirements.txt
-a----         3/23/2023   8:04 AM        1048576 test.txt
```
Write-manifest command:
```
PS C:\Users\User> rsconnect write-manifest voila .\voila\ .\voila\test.txt
Checking arguments...                            [ERROR]
Error: Could not find file .\voila\test.txt under .\voila
```

**Review by mmarchetti — APPROVED** [review]

**Comment by bcwu** [comment]
I am rolling back some of the changes because a Windows fix is larger than the scope of this issue.

**Comment by kgartland-rstudio** [comment]
This works for APIs and other content when write-manifest requires a directory but when a file is required like in `write-manifest notebook...` it still fails:

version:
```
> rsconnect version
1.16.1.dev117+g2adcec5
```
write-manifest command:
```
> rsconnect write-manifest notebook stock-report-jupyter/stock-report-jupyter.ipynb stock-report-jupyter/quandl-wiki-tsla.json.gz stock-report-jupyter/testfolder/test.txt --overwrite
Checking arguments...                            [OK]
Inspecting Python environment...                 [OK]
Creating manifest.json...                        [ERROR]
Error: ../quandl-wiki-tsla.json.gz must be under stock-report-jupyter.
```

It continues to work correctly within the target directory.

**Comment by bcwu** [comment]
include extra file checklist:
- [ ] works from outside the deployment directory
  - [ ] Voila
  - [ ] Jupyter notebook
  - [ ] flask, fastapi. streamlit, dash, bokeh, shiny
  - [ ] Quarto
- [ ] works from inside the deployment directory (i.e. ./)
  - [ ] Voila
  - [ ] Jupyter notebook
  - [ ] flask, fastapi. streamlit, dash, bokeh, shiny
  - [ ] Quarto

**Comment by kgartland-rstudio** [comment]
Tested all above content types on MacOS with and without extra files ✅ 
On Windows I ensured write-manifest inside / outside of the target directory continues to work for all content types. ✅

**Pull request merged**